src/morphology/morphoHelpers/suffixHelpers.js

```diff
@@ -0,0 +1,27 @@
+import { flattenDeep } from "lodash-es";
+
+/**
+ * Creates word forms from a list of given suffixes and a stem.
+ *
+ * @param {string}      stem                The stem on which to apply the suffixes.
+ * @param {string[]}    suffixes            The suffixes to apply.
+ * @param {string}      [appendToStem=""]   Optional material to append between the stem and the suffixes.
+ *
+ * @returns {string[]} The suffixed verb forms.
+ */
+export function applySuffixesToStem( stem, suffixes, appendToStem = "" ) {
+	return suffixes.map( suffix => stem.concat( appendToStem ).concat( suffix ) );
+}
+
+/**
+ * Creates word forms by appending all given suffixes on each of the given stems.
+ *
+ * @param {string[]}    stems               The stems on which to apply the suffixes.
+ * @param {string[]}    suffixes            The suffixes to apply.
+ * @param {string}      [appendToStem=""]   Optional material to append between the stem and the suffixes.
+ *
+ * @returns {string[]} The suffixed verb forms.
+ */
+export function applySuffixesToStems( stems, suffixes, appendToStem = "" ) {
+	return flattenDeep( stems.map( stem => applySuffixesToStem( stem, suffixes, appendToStem ) ) );
```

Comment:
I propose replacing this function with this:
```js
return stems.reduce( ( stem, list ) => {
  const stemWithSuffixes = applySuffixesToStem( stem, suffixes, appendToStem );
  return list.concat( stemsWithSuffixes );
}, [] );
```
This would loop over the list once, without the need of a second `flattenDeep` loop.
(This would only work when `stemsWithSuffixes` is always one level deep, but my guess is it is).